R/organize_data.R

```diff
@@ -0,0 +1,114 @@
+#' organize_data()
+#' Extracts data from objects to use in a shiny app.
+#'
+#' This function joins the result of [tune::fit_resamples()]  to the original dataset
+#' to give a dataframe that can be a Shiny input.
+#' @param x The [tune::fit_resamples()] result.
+#' @param ... Other parameters not currently used.
+#' @keywords models,  regression, graphs, classes, classif
+#' @export
+#' @return
+#' A data frame.
+organize_data <- function(x, ...) {
+  UseMethod("organize_data")
+}
+
+#' @export
+#' @rdname organize_data
+organize_data.default <- function(x, ...) {
+  rlang::abort("No `organize_data()` exists for this type of object.")
+}
+
+#' @export
+#' @rdname organize_data
+organize_data.tune_results <-
+  function(x,
+           ...) {
+    # TODO
+    original_data <- x$splits[[1]]$data
+    if (!(".predictions" %in% colnames(x))) {
+      rlang::abort(
+        paste0(
+          "The `.predictions` column does not exist. ",
+          "Refit with the control argument `save_pred = TRUE` to save predictions."
+        )
+      )
+    }
+    y_name <- tune::.get_tune_outcome_names(x)
+    if (!(y_name %in% names(original_data))) {
+      rlang::abort(glue::glue("'{y_name}' is not a column in the orignal data"))
+    }
+    sample_predictions <-
+      tune::collect_predictions(x, summarize = TRUE)
+    if (is.numeric(original_data[[y_name]]) == TRUE) {
+      sample_predictions <- sample_predictions %>%
+        dplyr::mutate(.residual = !!rlang::sym(y_name) - .pred)
+    }
+    preds <- sample_predictions %>%
+      dplyr::inner_join(original_data %>%
+        parsnip::add_rowindex() %>%
+        dplyr::select(-!!rlang::sym(y_name)),
+      by = ".row"
+      )
+    app_type <- get_app_type(original_data[[y_name]])
+    new_shiny_data(preds, y_name, app_type)
+  }
+# ------------------------------------------------------------------------------
+
+new_shiny_data <- function(predictions, y_name, subclass) {
+  if (!inherits(predictions, "data.frame")) {
+    rlang::abort("predictions should be a data frame")
+  }
+  if (nrow(predictions) == 0) {
+    rlang::abort("there should be at least one row of predictions")
+  }
+  # check to see if the right prediction columns are there
+  # TODO
+  # Think about other ways things could go wrong
+  if (!is.character(y_name)) {
+    rlang::abort("y_name should be a character string")
+  }
+  res <- list(predictions = predictions, y_name = y_name)
+  result <- structure(res, class = c(paste0(subclass, "_shiny_data"), "shiny_data"))
+  attr(result, "app_type") <- subclass
```

Comment:
These three lines are not required. Each of these pieces of information are in the output list.

Reply:
Wait I am confused here, I think this is the first time we create the list and assign its attributes. So we will need these lines? Without these lines, there is no `result` object, which is required for rest of the code.